operator/controllers/resources/suite_test.go

```diff
@@ -200,3 +200,38 @@ func validRandomResourceName(prefix string) string {
 	}
 	return prefix + string(b)
 }
+
+type testSetup struct {
+	tClient   auth.ClientI
+	k8sClient kclient.Client
+	namespace *core.Namespace
+}
+
+func setupKubernetesAndTeleport(t *testing.T, ctx context.Context) testSetup {
+	teleportServer, operatorName := defaultTeleportServiceConfig(t)
+
+	require.NoError(t, teleportServer.Start())
+
+	tClient := clientForTeleport(t, teleportServer, operatorName)
+	k8sClient := startKubernetesOperator(t, tClient)
+
+	ns := createNamespaceForTest(t, k8sClient)
+	return testSetup{tClient: tClient, k8sClient: k8sClient, namespace: ns}
+}
+
+func teleportCreateDummyRole(ctx context.Context, t *testing.T, roleName string, tClient auth.ClientI) error {
```

Comment:
nit: Let's be consistent with the order of context.Context and testing.T between this and setupKubernetesAndTeleport.